libraries/AP_ROMFS/AP_ROMFS.cpp

```diff
@@ -137,7 +137,7 @@ const char *AP_ROMFS::dir_list(const char *dirname, uint16_t &ofs)
     const size_t dlen = strlen(dirname);
     for ( ; ofs < ARRAY_SIZE(files); ofs++) {
         if (strncmp(dirname, files[ofs].filename, dlen) == 0 &&
-            files[ofs].filename[dlen] == '/') {
+            (dlen == 0 || files[ofs].filename[dlen] == '/')) {
```

Comment:
You should just early return if dlen is zero - the whole loop is pointless in that case

Reply:
Not my reading.  This just makes reading "" the same as reading "/"

Reply:
This patch is no longer in this PR.